### ISSUE TITLE
Postgres helm chart deploy endless loop

### ISSUE THREAD

**Issue by milanpro** [garden-io/garden#1381]
### Current Behavior

Using this module:

```yaml
kind: Module
description: Helm chart for the database
type: helm
name: postgres
chart: stable/postgresql
releaseName: db
values:
  postgresqlPassword: postgres
  postgresqlUsername: postgres
```
results in a loop

```
ℹ postgres                  → Deploying version v-9f9d813360...

...

   Kubernetes: Getting API resource info for group apps/v1beta2
   GET https://192.168.64.4:8443/apis/apps/v1beta2
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/secrets/db-postgresql
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql-headless
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql
   Fetching Kubernetes resource v1/Secret/db-postgresql
   Fetching Kubernetes resource v1/Service/db-postgresql-headless
   Fetching Kubernetes resource v1/Service/db-postgresql
   Fetching Kubernetes resource apps/v1beta2/StatefulSet/db-postgresql
   Kubernetes: Getting API resource info for group apps/v1beta2
   GET https://192.168.64.4:8443/apis/apps/v1beta2
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/secrets/db-postgresql
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql-headless
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql
   Fetching Kubernetes resource v1/Secret/db-postgresql
   Fetching Kubernetes resource v1/Service/db-postgresql-headless
   Fetching Kubernetes resource v1/Service/db-postgresql
   Fetching Kubernetes resource apps/v1beta2/StatefulSet/db-postgresql
   Kubernetes: Getting API resource info for group apps/v1beta2
   GET https://192.168.64.4:8443/apis/apps/v1beta2
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/secrets/db-postgresql
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql-headless
   GET https://192.168.64.4:8443/api/v1/namespaces/mpci/services/db-postgresql
✖ postgres                  → Deploying version v-9f9d813360...
Emit event 'taskError'

Failed deploying service 'postgres' (from module 'postgres'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Timed out waiting for postgres to deploy
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```
### Expected behavior

Since the database started successfully this should succeed at some point. (The next deploy will work using the already running database)

### Reproducible example

see above

### Your environment

- macOS Catalina 10.15.1
- garden 0.10.15
- kubectl:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-14T04:24:29Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
- docker
```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:34 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by edvald** [comment]
Hey @milanpro, thanks for the report! We can take a look. At first gloss the example doesn't result in the same issue on my side though. Could you paste a log where you run the deploy command with `-l=5`? This should print out which resources don't match when performing the rollout status check.

**Comment by milanpro** [comment]
Here is a longer log. The last part is the beginning of the loop.

```
Resolving provider local-kubernetes
Calling getServiceStatus handler for service postgres
Fetching chart...
Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube init --client-only' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
Calling 'configure' handler for 'helm'
Resolving provider local-kubernetes
Called 'configure' handler for 'helm'
   Syncing 0 files from services/postgres to .garden/build/postgres (with delete)
   File list: []
   Rsync args: -rptgo --ignore-missing-args --temp-dir /Users/milanpro/Projects/mpci/.garden/build/postgres.tmp --prune-empty-dirs --include-from=- --exclude=* --delete-excluded /Users/milanpro/Projects/mpci/services/postgres/ /Users/milanpro/Projects/mpci/.garden/build/postgres
ℹ postgres                  → Getting build status for v-d0cccc1c04...
Getting 'getBuildStatus' handler for module 'postgres' (type 'helm')
Resolving provider _default
Calling getBuildStatus handler for module postgres
ℹ postgres                  → Building version v-d0cccc1c04...
   Getting 'build' handler for module 'postgres' (type 'helm')
Resolving provider local-kubernetes
   Calling build handler for module postgres
   Fetching chart...
   Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube init --client-only' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube fetch stable/postgresql --destination /Users/milanpro/Projects/mpci/.garden/build/postgres --untar' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
   Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube fetch stable/postgresql --destination /Users/milanpro/Projects/mpci/.garden/build/postgres --untar' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
Preparing chart...
Writing chart values to /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql/garden-values.yml
Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube template --name db --namespace mpci --values /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql/garden-values.yml /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
   Preparing chart...
   Writing chart values to /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql/garden-values.yml
✔ postgres                  → Building version v-d0cccc1c04... → Done (took 3.2 sec)
Emit event 'taskComplete'
✔ tasks                     → Completed task build.postgres.1f96a9a0-16e9-11ea-88ae-0d98d92d8caf (took 3.387 sec)
ℹ Remaining tasks 2
Getting the release status for db
Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube status db --output json' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
Emit event 'taskComplete'
✔ tasks                     → Completed task get-service-status.postgres.1fabdf50-16e9-11ea-88ae-0d98d92d8caf (took 3.974 sec)
ℹ Remaining tasks 1
ℹ tasks                     → Processing task deploy.postgres.1f96d0b0-16e9-11ea-88ae-0d98d92d8caf
Currently in progress [deploy.postgres.1f96d0b0-16e9-11ea-88ae-0d98d92d8caf]
Resolving version for module postgres
Emit event 'taskProcessing'
Resolving version for module postgres
ℹ postgres                  → Deploying version v-d0cccc1c04...
   Getting deployService handler for service postgres
Resolving provider local-kubernetes
   Calling deployService handler for service postgres
   Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube template --name db --namespace mpci --values /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql/garden-values.yml /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
   Getting the release status for db
   Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube status db --output json' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
   Installing Helm release db
   Execing '/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube install /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql --name db --atomic --namespace mpci --timeout 300 --values /Users/milanpro/Projects/mpci/.garden/build/postgres/postgresql/garden-values.yml' in /Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64
   ℹ postgres                  → Waiting for resources to be ready...
   Fetching Kubernetes resource v1/Secret/db-postgresql
   Fetching Kubernetes resource v1/Service/db-postgresql-headless
   Fetching Kubernetes resource v1/Service/db-postgresql
   Fetching Kubernetes resource apps/v1beta2/StatefulSet/db-postgresql
   Kubernetes: Getting API resource info for group apps/v1beta2
   GET https://192.168.64.5:8443/apis/apps/v1beta2
   GET https://192.168.64.5:8443/api/v1/namespaces/mpci/secrets/db-postgresql
   GET https://192.168.64.5:8443/api/v1/namespaces/mpci/services/db-postgresql-headless
   GET https://192.168.64.5:8443/api/v1/namespaces/mpci/services/db-postgresql
   Fetching Kubernetes resource v1/Secret/db-postgresql
   Fetching Kubernetes resource v1/Service/db-postgresql-headless
   Fetching Kubernetes resource v1/Service/db-postgresql
   Fetching Kubernetes resource apps/v1beta2/StatefulSet/db-postgresql
```

**Comment by edvald** [comment]
Hmm. Generally we'd see some info about which resources are missing or out of sync, so I wonder why that's not coming up. What does `/Users/milanpro/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace mpci --kube-context minikube status db --output json` output?

**Comment by milanpro** [comment]
```json
{
  "name": "db",
  "info": {
    "status": {
      "code": 1,
      "resources": "==> v1/Pod(related)\nNAME              READY   STATUS    RESTARTS   AGE\ndb-postgresql-0   1/1     Running   0          38s\n\n==> v1/Secret\nNAME            TYPE     DATA   AGE\ndb-postgresql   Opaque   1      38s\n\n==> v1/Service\nNAME            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE\ndb-postgresql   ClusterIP   10.107.16.24   <none>        5432/TCP   38s\ndb-postgresql-headless   ClusterIP   None   <none>   5432/TCP   38s\n\n==> v1/StatefulSet\nNAME            READY   AGE\ndb-postgresql   1/1     38s\n\n",
      "notes": "** Please be patient while the chart is being deployed **\n\nPostgreSQL can be accessed via port 5432 on the following DNS name from within your cluster:\n\n    db-postgresql.mpci.svc.cluster.local - Read/Write connection\n\nTo get the password for \"postgres\" run:\n\n    export POSTGRES_PASSWORD=$(kubectl get secret --namespace mpci db-postgresql -o jsonpath=\"{.data.postgresql-password}\" | base64 --decode)\n\nTo connect to your database run the following command:\n\n    kubectl run db-postgresql-client --rm --tty -i --restart='Never' --namespace mpci --image docker.io/bitnami/postgresql:11.6.0-debian-9-r0 --env=\"PGPASSWORD=$POSTGRES_PASSWORD\" --command -- psql --host db-postgresql -U postgres -d postgres -p 5432\n\n\n\nTo connect to your database from outside the cluster execute the following commands:\n\n    kubectl port-forward --namespace mpci svc/db-postgresql 5432:5432 &\n    PGPASSWORD=\"$POSTGRES_PASSWORD\" psql --host 127.0.0.1 -U postgres -d postgres -p 5432\n"
    },
    "first_deployed": {
      "seconds": 1575538245,
      "nanos": 470431758
    },
    "last_deployed": {
      "seconds": 1575538245,
      "nanos": 470431758
    },
    "Description": "Install complete"
  },
  "namespace": "mpci"
}
```
This is the output while garden tries to deploy the Postgres

**Comment by edvald** [comment]
I think this issue may be fixed in v0.11. @milanpro have you tried a recent version?

**Comment by milanpro** [comment]
Yes, the problem seems to be fixed. Thank you!
